### PR TITLE
workflows: Zephyr workflow uses new 0.16.1 Zephyr SDK

### PR DESCRIPTION
**Pull request by aborisovich**
Updated Zephyr workflow - installed new Zephyr SDK v0.16.1 that is required by Zephyr main branch.
New Zephyr SDK is **NOT** backward compatible to older Zephyr revisions so the upgrade in `west.yml` is done to the commit: https://github.com/zephyrproject-rtos/zephyr/commit/9fc99928caf0 that requires this toolchain.
Build-windows job changes:
* new Zephyr SDK had changed extension and now is compressed using 7z format.
* from build-windows workflow removed unzip package that is no longer required by the setup.cmd Zephyr SDK installation script

Build-linux job changes:
* updated script `zephyr\docker-run.sh` that clones Zephyr Docker container to clone latest tag versioned v0.26.4.
* new container contains Zephyr SDK v0.16.1
* upgraded runner OS to Ubuntu 22.04 to match OS from Docker container

Sparse-Zephyr workflow:
* upgraded runner OS to Ubuntu 22.04 to match OS from the Zephyr workflow.

Signed-off-by: Andrey Borisovich <andrey.borisovich@intel.com>

### CODE REVIEW

**Comment by aborisovich** [comment]
@marc-hb we got all green on the Zephyr workflows.
Internal Intel CI had not finished yet but I think no point waiting for those.
This was with west.yml set to Zephyr main branch (what checks build-windows with zmain revision).
Dropping DNM and zmain revision from west.yml, let's see whether current SOF manifest rev works ok with new SDK.

**Review by marc-hb — COMMENTED** [review]
Looks great, thank you! Very glad it seems to have gone smoothly.

Just to be on the safe side please add this:

```diff
--- a/.github/workflows/sparse-zephyr.yml
+++ b/.github/workflows/sparse-zephyr.yml
@@ -15,9 +15,9 @@ jobs:
   # sof/scripts/parse_sparse_output.sh
   warnings-subset:
 
-    # We're sharing binaries with the zephyr-build container so keep
+    # We're sharing the sparse binary with the zephyr-build container so keep
     # this in sync with it.
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
 
     strategy:
       fail-fast: false
```

**Comment by aborisovich** [comment]
Wow - codestyle / checkpatch failed with the following message:

> WARNING: A patch subject line should describe the change not the tool that found it
> #4: 
> Subject: [PATCH] workflows: Sparse Zephyr upgraded runner OS to Ubuntu 22.04

I sometimes think this checkpatch is chatGTP powered or something...

**Comment by plbossart** [comment]
> Wow - codestyle / checkpatch failed with the following message:
> 
> > WARNING: A patch subject line should describe the change not the tool that found it
> > #4:
> > Subject: [PATCH] workflows: Sparse Zephyr upgraded runner OS to Ubuntu 22.04
> 
> I sometimes think this checkpatch is chatGTP powered or something...

it's been that way for a very long time...

**Comment by marc-hb** [comment]
checkpatch.pl is a Linux kernel tool. The Linux kernel tool has no CI scripts or configuration in the same git repo, nothing like the `.github/` directory or similar. So these false positives are rare in the kernel.

While this warning is wrong in this case, it is unfortunately very common to see people submitting bad commit messages like "fix warning from tool X" without describing what the actual code change is. Then this warning is very useful.

**Comment by marc-hb** [comment]
Also: https://www.kernel.org/doc/html/latest/dev-tools/checkpatch.html
> Checkpatch is not always right. Your judgement takes precedence over checkpatch messages. If your code looks better with the violations, then its probably best left alone.

False positives are OK because checkpatch only check PATCHES which means  warnings do NOT stick after merge like with other linters.

**Comment by marc-hb** [comment]
Thanks again @aborisovich , this solves a big problem!

Something's not right in 
https://github.com/thesofproject/sof/actions/runs/5017216786/jobs/8995052856?pr=7645, sparse compilation still looks for the old toolchain. I will look at it after a few meetings.


```
CMake Error at /zep_workspace/zephyr/cmake/compiler/gcc/target.cmake:12 (message):
  C compiler
  /home/cwd_user/zephyr-sdk-0.16.1/xtensa-intel_s1000_zephyr-elf/bin/xtensa-intel_s1000_zephyr-elf-gcc
  not found - Please check your toolchain installation
```

**Comment by marc-hb** [comment]
>  I will look at it after a few meetings.

OK this didn't take long: @aborisovich can you please bump sof/west.yml a bit further to https://github.com/zephyrproject-rtos/zephyr/commit/9fc99928caf0 ? We need that one too for MTL

**Review by kv2019i — APPROVED** [review]
Thanks @aborisovich !

**Comment by kv2019i** [comment]
The manifest mtl fails to 

https://github.com/thesofproject/sof/actions/runs/5017216783/jobs/8995053803?pr=7645
/home/cwd_user/zephyr-sdk-0.16.1/xtensa-intel_s1000_zephyr-elf/bin/xtensa-intel_s1000_zephyr-elf-gcc

While main mtl test passes and correct compiler is found:

https://github.com/thesofproject/sof/actions/runs/5017216783/jobs/8995053803?pr=7645
-- Found assembler: /home/cwd_user/zephyr-sdk-0.16.1/xtensa-intel_ace15_mtpm_zephyr-elf/bin/xtensa-intel_ace15_mtpm_zephyr-elf-gcc

Why is this happening, the PR is updating the manifest, right...?

**Comment by aborisovich** [comment]
All green now, thanks @marc-hb .
Let's wait for Intel Internal CI results and merge if positive.

**Review by tmleman — APPROVED** [review]

**Review by marc-hb — APPROVED** [review]
Great job, thanks a lot @aborisovich ! Please merge this first thing on Monday, thanks!

https://sof-ci.01.org/sofpr/PR7645/build8158/devicetest/index.html?model=TGLU_RVP_NOCODEC_IPC4ZPH&testcase=multiple-pause-resume-5 has the known corrupted heap issue 7191

https://sof-ci.01.org/sofpr/PR7645/build8157/devicetest is all green.

Checkpatch is wrong.

Everything else is green!

**Pull request merged**
